sdks/python/apache_beam/dataframe/frame_base.py

```diff
@@ -160,8 +160,10 @@ def _proxy_method(
     restrictions=None,
     inplace=False,
     base=None,
-    requires_partition_by=partitionings.Singleton(),
-    preserves_partition_by=partitionings.Arbitrary()):
+    *,
+    requires_partition_by,  # type: partitionings.Partitioning
+    preserves_partition_by,  # type: partitionings.Partitioning
```

Comment:
should we prefer PEP 484 type hints now?

Reply:
Good point, though I think this cleanup should be done separately (and prefer to be consistent within a method/file).